Ginger/Ginger/BusinessFlowPages_New/AddActionMenu/RecordNavPage.xaml.cs

```diff
@@ -40,197 +27,289 @@ namespace Ginger.BusinessFlowsLibNew.AddActionMenu
+            {
+                mContext = (Context)sender;
+                xWinGridUC.mContext = mContext;
+                InitMethods();
+                if (IsRecording)
+                {
+                    IsRecording = !IsRecording;
+                    StopRecording();
+                    SetRecordingButtonText();
+                }
+            }
+        }
+
+        /// <summary>
+        /// This method will set the page to default
+        /// </summary>
+        /// <param name="context"></param>
+        public void SetDefault(Context context)
+        {
+            if(mActParentActivity != context.BusinessFlow.CurrentActivity)
+            {                
+                xWinGridUC.UpdateWindowsList();
+                xIntegratePOM.IsChecked = false;
+            }
+            mActParentActivity = context.BusinessFlow.CurrentActivity;
+
+            if (AgentHelper.Instance.CheckIfAgentIsRunning(context))
+            {
+                mWindowExplorerDriver = AgentHelper.Instance.WindowExplorerDriver;
+                xWinGridUC.mWindowExplorerDriver = mWindowExplorerDriver;
+                xStartAgentButton.IsEnabled = false;
+            }
+            else
+            {
+                xStartAgentButton.IsEnabled = true;
+            }
+        }
+
+        public void SetMultiplePropertiesGridView()
         {
-            isRecording = !isRecording;
-            UpdateUI();
-            //stop Live Spy
-            //if (LiveSpyButton.IsChecked == true)
-            //    if (dispatcherTimer != null)
-            //        dispatcherTimer.IsEnabled = false;
+            gridPOMListItems.SetTitleLightStyle = true;
+            GridViewDef view = new GridViewDef(GridViewDef.DefaultViewName);
+            view.GridColsView = new ObservableList<GridColView>();
+            view.GridColsView.Add(new GridColView() { Field = nameof(POMBindingObjectHelper.ItemName), Header = "Name", WidthWeight = 120, AllowSorting = true, BindingMode = BindingMode.OneWay, ReadOnly = true });
+            view.GridColsView.Add(new GridColView() { Field = nameof(POMBindingObjectHelper.ContainingFolder), Header = "ContainingFolder", WidthWeight = 150, AllowSorting = true, BindingMode = BindingMode.OneWay, ReadOnly = true });
+            gridPOMListItems.btnAdd.Click += BtnAdd_Click;
+            gridPOMListItems.SetAllColumnsDefaultView(view);
+            gridPOMListItems.InitViewItems();
+        }
+        
+        private void BtnAdd_Click(object sender, RoutedEventArgs e)
+        {
+            ApplicationPOMsTreeItem appModelFolder;
+
+            RepositoryFolder<ApplicationPOMModel> repositoryFolder = WorkSpace.Instance.SolutionRepository.GetRepositoryItemRootFolder<ApplicationPOMModel>();
+            appModelFolder = new ApplicationPOMsTreeItem(repositoryFolder);
+            mApplicationPOMSelectionPage = new SingleItemTreeViewSelectionPage("Page Objects Model Element", eImageType.ApplicationPOMModel, appModelFolder,
```

Comment:
don't create new selection tree page each time, use same one if already exist so user won't need to expand all root folder each time he adds POM

Reply:
Done